test/reporters/appinsights.go

```diff
@@ -0,0 +1,63 @@
+// Package reporters implements gingo Reporter for  Azure App Insights
+package reporters
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/onsi/ginkgo/config"
+	"github.com/onsi/ginkgo/types"
+
+	"github.com/openshift/openshift-azure/pkg/util/azureclient/insights"
+)
+
+type AzureAppInsightsReporter struct {
+	icli insights.TelemetryClient
+}
+
+func NewAzureAppInsightsReporter() *AzureAppInsightsReporter {
+	icli := insights.NewTelemetryClient(os.Getenv("AZURE_APP_INSIGHTS_KEY"))
+	icli.Context().CommonProperties["type"] = "gotest"
```

Comment:
suggest s/gotest/ginkgo/ - we will probably want another reporter like this for `go test` in the future